libraries/AP_Logger/LogFile.cpp

```diff
@@ -178,25 +178,68 @@ void AP_Logger::Write_RCIN(void)
 // Write an SERVO packet
 void AP_Logger::Write_RCOUT(void)
 {
-    const struct log_RCOUT pkt{
-        LOG_PACKET_HEADER_INIT(LOG_RCOUT_MSG),
-        time_us       : AP_HAL::micros64(),
-        chan1         : hal.rcout->read(0),
-        chan2         : hal.rcout->read(1),
-        chan3         : hal.rcout->read(2),
-        chan4         : hal.rcout->read(3),
-        chan5         : hal.rcout->read(4),
-        chan6         : hal.rcout->read(5),
-        chan7         : hal.rcout->read(6),
-        chan8         : hal.rcout->read(7),
-        chan9         : hal.rcout->read(8),
-        chan10        : hal.rcout->read(9),
-        chan11        : hal.rcout->read(10),
-        chan12        : hal.rcout->read(11),
-        chan13        : hal.rcout->read(12),
-        chan14        : hal.rcout->read(13)
-    };
-    WriteBlock(&pkt, sizeof(pkt));
+    const uint32_t enabled_mask = ~SRV_Channels::get_output_channel_mask(SRV_Channel::k_GPIO);
+
+    if ((enabled_mask & 0x3FFF) != 0) {
+        const struct log_RCOUT pkt{
+            LOG_PACKET_HEADER_INIT(LOG_RCOUT_MSG),
+            time_us       : AP_HAL::micros64(),
+            chan1         : hal.rcout->read(0),
+            chan2         : hal.rcout->read(1),
+            chan3         : hal.rcout->read(2),
+            chan4         : hal.rcout->read(3),
+            chan5         : hal.rcout->read(4),
+            chan6         : hal.rcout->read(5),
+            chan7         : hal.rcout->read(6),
+            chan8         : hal.rcout->read(7),
+            chan9         : hal.rcout->read(8),
+            chan10        : hal.rcout->read(9),
+            chan11        : hal.rcout->read(10),
+            chan12        : hal.rcout->read(11),
+            chan13        : hal.rcout->read(12),
+            chan14        : hal.rcout->read(13)
+        };
+        WriteBlock(&pkt, sizeof(pkt));
+    }
+
+#if NUM_SERVO_CHANNELS >= 15
+    if ((enabled_mask & 0x3C000) != 0) {
+        const struct log_RCOUT2 pkt2{
+            LOG_PACKET_HEADER_INIT(LOG_RCOUT2_MSG),
```

Comment:
Not a blocker, but we've tended to use the message name in these.  So `log_RCO2` and `LOG_RCO2_MSG`